File: leetcode/C++/transform-to-chessboard.cpp
class Solution {
public:
    int movesToChessboard(vector<vector<int>>& board) {
        int sz=board.size(), rn=0, cn=0, rr=0, cr=0;
        for(int i=0; i<sz; ++i){
            for(int j=0; j<sz; ++j){
                if((board[0][0]^board[i][j]^board[i][0]^board[0][j])) return -1;
            }
        }
        for(int i=0; i<sz; ++i){
            rn+=board[0][i]; cn+=board[i][0];
            if((i&1)==board[0][i]) ++rr;
            if((i&1)==board[i][0]) ++cr;
        }
        if(!(rn==sz/2 || rn==(sz+1)/2)) return -1;
        if(!(cn==sz/2 || cn==(sz+1)/2)) return -1;
        if(sz&1){
            if(rr&1) rr=sz-rr;
            if(cr&1) cr=sz-cr;
        }
        else{
            rr=min(sz-rr,rr);
            cr=min(sz-cr,cr);
        }
        return (rr+cr)/2;
    }
};

File: leetcode/Python/transform-to-chessboard.py
class Solution:
    def movesToChessboard(self, board: List[List[int]]) -> int:
        sz, r, c, rr, cr = len(board), 0, 0, 0, 0
        for i in range(sz):
            for j in range(sz):
                if (board[0][0]^board[i][0]^board[0][j]^board[i][j]) == 1:
                    return -1
        for i in range(sz):
            r+=board[0][i]
            c+=board[i][0]
            rr+=1 if (i&1)==board[0][i] else 0
            cr+=1 if (i&1)==board[i][0] else 0
        if not (r==sz//2 or r==(sz+1)//2) or not (c==sz//2 or c==(sz+1)//2):
            return -1
        if (sz&1)==1:
            rr = sz-rr if (rr&1)==1 else rr
            cr = sz-cr if (cr&1)==1 else cr
        else:
            rr = min(rr,sz-rr)
            cr = min(cr,sz-cr)
        return (rr+cr)//2